### PR TITLE
Custom Reports SQL query fixes

### LINKED ISSUE #14704: [Bug]: Custom Reports SQL removes line breaks too greedily
### Expected behavior

It is sometimes necessary to have linebreaks `\n` in the query in Custom Reports, they should be preserved.

### Actual behavior

They appear to be greedily removed in `models/Tool/CustomReport/Adapter/Sql.php`

### Steps to reproduce

Add a Custom Report query with e.g. `WHERE object_product.shortDescription LIKE "%\n\n%"`, it should return all rows where there is two consecutive linebreaks, but instead returns a report where shortDescription is _anything_.

### PR DESCRIPTION
**Pull request by NiklasBr**
## Changes in this pull request  
Resolves #14704
- remove `str_replace` because it interfered with some conditions
- add line-breaks to make debugging a bit easier, also allows us to use `-- SQL comments in the report input areas`

### CODE REVIEW

File: models/Tool/CustomReport/Adapter/Sql.php [hunk 1]
```diff
@@ -91,9 +91,9 @@ protected function buildQueryString($config, $ignoreSelectAndGroupBy = false, $d
         $sql = '';
         if (!empty($config['sql']) && !$ignoreSelectAndGroupBy) {
             if (strpos(strtoupper(trim($config['sql'])), 'SELECT') !== 0) {
```
Comment (jdreesen) [inline models/Tool/CustomReport/Adapter/Sql.php]:
```suggestion
            if (!str_starts_with(strtoupper(trim($config['sql'])), 'SELECT')) {
```

File: models/Tool/CustomReport/Adapter/Sql.php [hunk 2]
```diff
@@ -102,23 +102,23 @@ protected function buildQueryString($config, $ignoreSelectAndGroupBy = false, $d
         }
         if (!empty($config['from'])) {
             if (strpos(strtoupper(trim($config['from'])), 'FROM') !== 0) {
```
Comment (jdreesen) [inline models/Tool/CustomReport/Adapter/Sql.php]:
```suggestion
            if (!str_starts_with(strtoupper(trim($config['from'])), 'FROM')) {
```

File: models/Tool/CustomReport/Adapter/Sql.php [hunk 3]
```diff
@@ -102,23 +102,23 @@ protected function buildQueryString($config, $ignoreSelectAndGroupBy = false, $d
...
         if (!empty($config['where'])) {
             if (str_starts_with(strtoupper(trim($config['where'])), 'WHERE')) {
                 $config['where'] = preg_replace('/^\s*WHERE\s*/', '', $config['where']);
             }
-            $sql .= ' WHERE (' . str_replace("\n", ' ', $config['where']) . ')';
+            $sql .= "\n" . 'WHERE (' . $config['where'] . ')';
         }
 
         if (!empty($config['groupby']) && !$ignoreSelectAndGroupBy) {
             if (strpos(strtoupper(trim($config['groupby'])), 'GROUP BY') !== 0) {
```
Comment (jdreesen) [inline models/Tool/CustomReport/Adapter/Sql.php]:
```suggestion
            if (!str_starts_with(strtoupper(trim($config['groupby'])), 'GROUP BY')) {
```

**Comment by mcop1** [comment]
Thank you 👍

**Pull request merged**